modules/database/export.bal

```diff
@@ -0,0 +1,485 @@
+        timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId);
+        timelineSteps.push(timelineStepExport);
+        foreach var substep in timelineStepExport.timelineSubsteps {
+            foreach var substepData in substep.substepDataList {
+                dataIdList.push(substepData.dataId);
+            }
+        }
+    }
+
+    // ignore duplicates in dataIdList and create experiment data list
+    int[] sortedDataIdList = dataIdList.sort();
+    int tmp = -1;
+    foreach int dataId in sortedDataIdList {
+        if dataId != tmp {
+            tmp = dataId;
+            ExperimentDataExport experimentData = check getExperimentDataExport(experimentId, dataId);
+            experimentDataList.push(experimentData);
+        }
+    }
+
+    return {
+        experiment: {name: experiment.name, description: experiment.description},
+        timelineSteps: timelineSteps,
+        experimentDataList: experimentDataList
+    };
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Export main functionality ///////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+# Prepare zip file for export of an experiment.
+#
+# + experimentId - experiment id of the new (cloned) experiment
+# + config - export configuration // TODO
+# + os - os type to determine appropriate exec command
+# + return - record with details about created zip files or error
+public isolated transactional function exportExperiment(int experimentId, string? config, string os) returns ExperimentExportZip|error {
+
+    // TODO: config
+
+    ExperimentCompleteExport experimentComplete = check getExperimentDBExport(experimentId);
+    // data files
+    string[] dataFileLocations = [];
+    ExperimentDataExport[] experimentDataList = experimentComplete.experimentDataList;
+    foreach var experimentData in experimentDataList {
+        string location = experimentData.location;
+
+        // remove path from file location
+        int? index = location.lastIndexOf("/");
+        if index is () {
+            index = location.lastIndexOf("\\");
+        }
+        if index is () {
+            return error("Unable to determine relative file location.");
+        } else {
+            experimentData.location = location.substring(index + 1, location.length()); // only file name
+            var abspath = check file:getAbsolutePath("experimentData/" + experimentId.toString() + "/" + experimentData.location);
+            log:printInfo("Add file " + abspath + "..."); // TODO: remove
+            dataFileLocations.push(abspath);
+        }
+    }
+
+    // string tmpDir = check file:createTempDir();
+    if !(check file:test("tmp", file:EXISTS)) {
+        check file:createDir("tmp", file:RECURSIVE);
+    }
+    json experimentCompleteJson = experimentComplete;
+    string jsonFile = "tmp/experiment.json";
```

Comment:
these paths should all contain the export/import db ID so that multiple exports running at the same time do not override each other (a small chance but could happen...)